monix-execution/jvm/src/main/scala/monix/execution/schedulers/TracingScheduler.scala

```diff
@@ -0,0 +1,89 @@
+package monix.execution.schedulers
+
+import java.util.concurrent.{ScheduledExecutorService, TimeUnit}
+import monix.execution.misc.Local
+import monix.execution.{Cancelable, Scheduler, UncaughtExceptionReporter, ExecutionModel => ExecModel}
+import scala.concurrent.ExecutionContext
+
+final class TracingScheduler private (
+                                       scheduler: ScheduledExecutorService,
+                                       ec: ExecutionContext,
+                                       r: UncaughtExceptionReporter,
+                                       val executionModel: ExecModel)
+  extends ReferenceScheduler with BatchingScheduler { self =>
+
+  override def executeAsync(r: Runnable): Unit =
+    executeWithTrace(r)
+
+  /** Executes the given task with tracing.
+    *
+    * @param r is the callback to be executed
+    */
+  def executeWithTrace(r: Runnable): Unit = {
```

Comment:
If `executeAsync` is an alias of `executeWithTrace`, then why does `executeWithTrace` have to exist at all?

Reply:
@alexandru you are right, I did that at the beginning while I was trying to understand the schedulers and how they are used in TaskRunLoop.